### PR TITLE
Prepare v0.89.0

### PR DESCRIPTION
**Pull request by jpkrohling**
Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: distributions/otelcol-contrib/manifest.yaml [hunk 1]
```diff
@@ -2,213 +2,201 @@ dist:
...
 connectors:
-  - gomod: go.opentelemetry.io/collector/connector/forwardconnector v0.88.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/countconnector v0.88.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/datadogconnector v0.88.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/routingconnector v0.88.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/servicegraphconnector v0.88.0
-  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/connector/spanmetricsconnector v0.88.0
-
-# When adding a replace, add a comment before it to document why it's needed and when it can be removed
-replaces:
```
Comment (jpkrohling) [inline distributions/otelcol-contrib/manifest.yaml:206]: It looks like we can get rid of all of them?

File: distributions/otelcol-contrib/manifest.yaml [hunk 2]
```diff
@@ -2,213 +2,203 @@ dist:
...
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/redactionprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/remotetapprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourceprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/routingprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/servicegraphprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanmetricsprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/spanprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/sumologicprocessor v0.89.0
```
Comment (jpkrohling) [inline distributions/otelcol-contrib/manifest.yaml:105]: From #423

File: distributions/otelcol-contrib/manifest.yaml [hunk 3]
```diff
@@ -2,213 +2,203 @@ dist:
...
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/deltatorateprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/filterprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/groupbyattrsprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/groupbytraceprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/k8sattributesprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricsgenerationprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/metricstransformprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/probabilisticsamplerprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/redactionprocessor v0.89.0
+  - gomod: github.com/open-telemetry/opentelemetry-collector-contrib/processor/remotetapprocessor v0.89.0
```
Comment (jpkrohling) [inline distributions/otelcol-contrib/manifest.yaml:98]: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27873

File: .goreleaser.yaml [hunk 4]
```diff
@@ -1,5 +1,3 @@
-partial:
-  by: target
```
Comment (mx-psi) [inline .goreleaser.yaml:2]: Why was this removed?

File: distributions/otelcol-contrib/manifest.yaml (hunk 1 above)
Comment (mx-psi) [reply distributions/otelcol-contrib/manifest.yaml:206]: I would prefer to do this on a separate PR and not during the release to avoid any last minute errors

File: .goreleaser.yaml (hunk 4 above)
Comment (jpkrohling) [reply .goreleaser.yaml]: Running locally, I saw a warning that goreleaser files were out of sync, so I ran `make goreleaser`. Given that I got an issue about the runner running out of space, I reverted this change for now.

**Review by mx-psi — APPROVED** [review]
